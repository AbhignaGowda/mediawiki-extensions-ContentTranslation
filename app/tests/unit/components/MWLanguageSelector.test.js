import { shallowMount, mount } from "@vue/test-utils";
import MwLanguageSelector from "@/lib/mediawiki.ui/components/MWLanguageSelector";

const languages = [
  {
    code: "en",
    name: "English"
  },
  {
    code: "ml",
    name: "മലയാളം"
  },
  {
    code: "es",
    name: "Espanol"
  }
];

describe("MWLanguageSelector.vue", () => {
  it("renders passed languages", () => {
    const componentClass = "mw-ui-language-selector";
    const placeholder = "Search for languages";
    const wrapper = shallowMount(MwLanguageSelector, {
      propsData: { placeholder, languages }
    });
    expect(wrapper.contains("mw-input-stub")).toBe(true);
    expect(wrapper.find("mw-input-stub").attributes("placeholder")).toMatch(
      placeholder
    );
    expect(wrapper.findAll(".mw-ui-language-selector__result").length).toBe(
      languages.length
    );
    expect(wrapper.classes(componentClass)).toBe(true);
    expect(wrapper.element).toMatchSnapshot();
  });

  it("render search results when queried", async () => {
    const componentClass = "mw-ui-language-selector";
    const wrapper = shallowMount(MwLanguageSelector, {
      propsData: { languages }
    });
    wrapper.find("mw-input-stub").vm.$emit("update", "Engl");
    expect(wrapper.vm.results).toStrictEqual([{ code: "en", name: "English" }]);
    await wrapper.vm.$nextTick();
    expect(wrapper.find("mw-input-stub").attributes("suggestion")).toBe(
      "English"
    );
    expect(wrapper.find("mw-input-stub").attributes("value")).toBe("Engl");
    expect(wrapper.findAll(".mw-ui-language-selector__result").length).toBe(1);
    expect(wrapper.element).toMatchSnapshot();
  });
});
