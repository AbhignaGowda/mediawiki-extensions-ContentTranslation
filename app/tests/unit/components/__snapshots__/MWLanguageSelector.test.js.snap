// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MWLanguageSelector.vue render search results when queried 1`] = `
<div
  class="mw-ui-language-selector row"
>
  <mw-input-stub
    class="mw-ui-language-selector__input col-12"
    icon="M7.5 13c3.04 0 5.5-2.46 5.5-5.5S10.54 2 7.5 2 2 4.46 2 7.5 4.46 13 7.5 13zm4.55.46A7.432 7.432 0 0 1 7.5 15C3.36 15 0 11.64 0 7.5S3.36 0 7.5 0C11.64 0 15 3.36 15 7.5c0 1.71-.57 3.29-1.54 4.55l6.49 6.49-1.41 1.41-6.49-6.49z"
    iconsize="24"
    indicatorsize="24"
    suggestion="English"
    tabindex="0"
    type="search"
    value="Engl"
  />
   
  <div
    class="mw-ui-language-selector__results col-12"
  >
    <div
      class="mw-ui-language-selector__result row"
    >
      <span
        class="mw-ui-language-selector__result-name col-10"
      >
        English
      </span>
       
      <span
        class="mw-ui-language-selector__result-code col-2"
      >
        en
      </span>
    </div>
  </div>
</div>
`;

exports[`MWLanguageSelector.vue renders passed languages 1`] = `
<div
  class="mw-ui-language-selector row"
>
  <mw-input-stub
    class="mw-ui-language-selector__input col-12"
    icon="M7.5 13c3.04 0 5.5-2.46 5.5-5.5S10.54 2 7.5 2 2 4.46 2 7.5 4.46 13 7.5 13zm4.55.46A7.432 7.432 0 0 1 7.5 15C3.36 15 0 11.64 0 7.5S3.36 0 7.5 0C11.64 0 15 3.36 15 7.5c0 1.71-.57 3.29-1.54 4.55l6.49 6.49-1.41 1.41-6.49-6.49z"
    iconsize="24"
    indicatorsize="24"
    placeholder="Search for languages"
    suggestion="English"
    tabindex="0"
    type="search"
    value=""
  />
   
  <div
    class="mw-ui-language-selector__results col-12"
  >
    <div
      class="mw-ui-language-selector__result row"
    >
      <span
        class="mw-ui-language-selector__result-name col-10"
      >
        English
      </span>
       
      <span
        class="mw-ui-language-selector__result-code col-2"
      >
        en
      </span>
    </div>
    <div
      class="mw-ui-language-selector__result row"
    >
      <span
        class="mw-ui-language-selector__result-name col-10"
      >
        മലയാളം
      </span>
       
      <span
        class="mw-ui-language-selector__result-code col-2"
      >
        ml
      </span>
    </div>
    <div
      class="mw-ui-language-selector__result row"
    >
      <span
        class="mw-ui-language-selector__result-name col-10"
      >
        Espanol
      </span>
       
      <span
        class="mw-ui-language-selector__result-code col-2"
      >
        es
      </span>
    </div>
  </div>
</div>
`;
